app/grandchallenge/components/views.py

```diff
@@ -84,3 +89,46 @@ def get_queryset(self):
 
     def get_result_label(self, result):
         return result.title
+
+
+class InterfaceProcessingMixin:
+    def process_data_for_object(self, data):
+        raise NotImplementedError
+
+    def form_invalid(self, form):
+        return JsonResponse(form.errors, status=400)
+
+    def form_valid(self, form):
+        self.process_data_for_object(form.cleaned_data)
+        return JsonResponse({"redirect": self.get_success_url()})

```

Comment:
This might seem strange, but we want to prevent the default from happening here since the form can contain extra, dynamically added fields (new interfaces) for which we want to be able to show errors as well. The default here would just return the form with the errors for the fields that were declared in the form init function. The fields that are added by the user would just disappear.

Reply:
Hmm, seems a bit anti-Django. Is there another approach? Could you append the existing fields on the form?

Reply:
Maybe I have found a way. Needs more work still, but I think I'm getting there.

Reply:
Got it to work, but I'm not sure if this implementation is 100% kosher either, and it does come with some drawbacks as well. I'll commit and see what you think. From a user perspective, it is a bit weird that the look of the form changes when a validation error is returned:

Before submitting the form with a new interface (number):
![image](https://github.com/comic/grand-challenge.org/assets/30069334/0978c24f-1e47-4c3b-8fcc-88efe964d7d5)

and after validation:

![image](https://github.com/comic/grand-challenge.org/assets/30069334/732583d1-dfd8-4cc9-8607-953eb5d75ce0)

The box around the new interface disappears and also the option to remove it.

Reply:
What happened to the submitted value and the metics json file widget?

Reply:
That the widget disappears is a seperate issue with the json editor, I will create a seperate PR (it's related to https://github.com/comic/grand-challenge.org/issues/3162). That the value of number disappears is something that I'm still trying to understand.

Reply:
Got the values to persist. Implementation complete, but I'm now wondering if using htmx to send along the new interfaces (instead of the custom js script that currently does this) might also work and be neater. We could use hx-include to include them. Any objections before I go down the potential rabbit whole of getting that to work?

Reply:
I don't see why we wouldn't use htmx, makes for a lot less code. I'll prepare a commit